packages/esm-form-entry-app/src/app/services/concept.service.ts

```diff
@@ -19,50 +22,39 @@ export class ConceptService {
 
   constructor(private http: HttpClient, private windowRef: WindowRef) {}
 
-  public getUrl(): string {
-    return this.windowRef.openmrsRestBase + 'concept';
-  }
-
-  public searchConceptByUUID(conceptUUID: string, lang: string): Observable<any> {
-    return this.http.get(this.getUrl() + `/${conceptUUID}?v=full&lang=${lang}`, {
-      headers: this.headers,
-    });
-  }
-
-  public searchBulkConceptsByUUID(conceptUuids: Array<string>, lang: string): Observable<Array<ConceptMetadata>> {
-    const observablesArray = [];
-    const relativeConceptLabelUrls = ConceptService.getRelativeConceptLabelUrls(conceptUuids, lang);
-
-    for (const relativeConceptLabelUrl of relativeConceptLabelUrls) {
-      observablesArray.push(
+  public searchConceptsByIdentifiers(conceptIdentifiers: Array<string>) {
+    return of(...ConceptService.getConceptReferenceUrls(conceptIdentifiers)).pipe(
+      map((referenceUrl) =>
         this.http
-          .get<ListResult<ConceptMetadata>>(this.windowRef.openmrsRestBase + relativeConceptLabelUrl, {
+          .get<ConceptReferencesResult>(this.windowRef.openmrsRestBase + referenceUrl + '?v=custom:(uuid,display)', {
             headers: this.headers,
           })
           .pipe(
-            map((r) => {
-              return r.results;
-            }),
+            map((result) =>
+              Object.entries(result).reduce((acc, reference) => {
+                acc.push({
+                  uuid: reference[0],
+                  display: reference[1].display,
+                });
+
+                return acc;
+              }, [] as ConceptMetadata[]),
```

Comment:
```suggestion
              }, [] as Array<ConceptMetadata>),
```

(Just for consistency with the rest of the codebase.)